src/TriggerBinding/SqlScalerProvider.cs

```diff
@@ -55,6 +75,59 @@ public ITargetScaler GetTargetScaler()
             return this._targetScaler;
         }
 
+        public static async Task<SqlScalerProvider> CreateAsync(IServiceProvider serviceProvider, TriggerMetadata triggerMetadata)
+        {
+            var provider = new SqlScalerProvider(serviceProvider, triggerMetadata);
+            await provider.InitializeAsync();
+            return provider;
+        }
+
+
+        public async Task InitializeAsync(CancellationToken cancellationToken = default)
+        {
+            using (var connection = new SqlConnection(this._connectionString))
+            {
+                await connection.OpenAsyncWithSqlErrorHandling(cancellationToken);
+                ServerProperties serverProperties = await GetServerTelemetryProperties(connection, this._logger, cancellationToken);
+                this._telemetryProps.AddConnectionProps(connection, serverProperties);
+                await VerifyDatabaseSupported(connection, this._logger, cancellationToken);
+
+                int userTableId = await GetUserTableIdAsync(connection, this._userTable, this._logger, cancellationToken);
```

Comment:
We had discussed NOT doing this here and instead having the scaler return the total count of all changes if the tables don't exist. This would cause it to scale up, which would create the actual instance that would then create the tables as normal.

Is there a reason you decided not to do that? I'm not thrilled about having table creation happen in multiple places in our code. It makes it more likely that if we ever make any changes to how they're created that we'll miss updating one or the other and cause problems.

Reply:
Yeah, it would also work.

You need to catch an exception here (schema does not exist, GlobalStateTable does not exists or GlobalStateTableRow does not exists):
https://github.com/Azure/azure-functions-sql-extension/blob/main/src/TriggerBinding/SqlTriggerTargetScaler.cs#L29

and return:
            return new TargetScalerResult
            {
                TargetWorkerCount = 1
            };
Then a worker will be assigned, and listener will create required stuff.

Reply:
I believe it’s better for the SC to attempt creating the GlobalState, since we only try once.
Otherwise, we might end up with worker 1 being permanently assigned, as the SC will keep voting for it if GlobalState creation fails for any reason.

Reply:
If the worker gets in a bad state (e.g. if global state table creation fails in the worker it will throw an exception and exit) will the scaler make a new one? If so then keeping 1 worker around seems like the behavior we want then - since it'll just keep retrying to make the table until it succeeds - which we need for the trigger to work correctly.

Reply:
we do not want to keep a worker around just waiting until the issue with creation is fixed as it would affect customer's bill

Reply:
If we want to scale out only once right after monitoring started for MSSQL trigger - yes, the bad worker will exit and will never add again until customer explicitly changed the function app. 

EventHubs have the same scenario - creating checkpoint blob to store position client position in the stream. The blob is created on SDK level by `EventProcessorClient`. There is no such thing as SQL SDK for monitoring changes, so the code lives in MSSQL azure functions extension.

Reply:
In general, I am ok with scale out 1 instance but only once